libringbuffer/src/ringbuffer.c

```diff
@@ -27,6 +27,10 @@ ringbuffer_t *rb_new(void *base, size_t size) {
     return r;
 }
 
+int rb_has_data(ringbuffer_t *r) {
+    return (r->base[r->offset] == 0);
```

Comment:
The change looks ok withing the current implementation behavior. But I wonder if `rb_new()`should do a `r->base[0] = 0` to start with an empty buffer - or it's a feature that there can be data in the buffer? Seem if the buffer does not contain a 0 then one can read over and over from the ring buffer as it never gets empty?

Reply:
I'm not sure. I didn't see any problem with that (and I malloc'd the buffers instead of calloc-ing them)

Maybe track that as a separate issue?